src/main/java/org/opensearch/ad/transport/DeleteAnomalyResultsTransportAction.java

```diff
@@ -61,7 +61,8 @@ protected void doExecute(Task task, DeleteByQueryRequest request, ActionListener
     }
 
     public void delete(DeleteByQueryRequest request, ActionListener<BulkByScrollResponse> listener) {
-        User user = getUserContext(client);
+        // Temporary null user for AD extension without security. Will always validate role.
+        UserIdentity user = getNullUser();
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
```

Comment:
We are not targeting threadpool for SDKClient. Does having threadContext here will impact the client?

Reply:
The removal of the thread pool will be done in a pending PR that I'll submit as soon as this one is merged.  See earlier similar comment and my response here: https://github.com/opensearch-project/anomaly-detection/pull/617#discussion_r929516077